File: src/main/java/io/curiousoft/ijudi/ordermanagement/repo/StoreRepository.java
package io.curiousoft.ijudi.ordermanagement.repo;

import io.curiousoft.ijudi.ordermanagement.model.StoreProfile;
import io.curiousoft.ijudi.ordermanagement.model.StoreType;

import java.util.Arrays;
import java.util.List;

public interface StoreRepository extends ProfileRepo<StoreProfile> {

    List<StoreProfile> findByFeatured(boolean b);

    List<StoreProfile> findByOwnerId(String ownerId);


    List<StoreProfile> findByLatitudeBetweenAndLongitudeBetween(double v, double v1, double v2, double v3);

    List<StoreProfile> findByLatitudeBetweenAndLongitudeBetweenAndStoreType(double minLat, double maxLat,
                                                                            double minLong, double maxLong,
                                                                            StoreType storeType);
}

File: izinga-commons/src/main/kotlin/commons/repo/StoreRepository.kt
package io.curiousoft.izinga.commons.repo

import io.curiousoft.izinga.commons.model.StoreProfile
import io.curiousoft.izinga.commons.model.StoreType
import java.util.*

interface StoreRepository : ProfileRepo<StoreProfile> {
    fun findByFeatured(b: Boolean): List<StoreProfile>?
    fun findByOwnerId(ownerId: String?): List<StoreProfile>?
    fun findByLatitudeBetweenAndLongitudeBetween(v: Double, v1: Double, v2: Double, v3: Double): List<StoreProfile>?
    fun findByLatitudeBetweenAndLongitudeBetweenAndStoreType(
        minLat: Double, maxLat: Double,
        minLong: Double, maxLong: Double,
        storeType: StoreType?
    ): List<StoreProfile?>?

    fun findOneByIdOrShortName(id: String?, shortname: String?): Optional<StoreProfile>?
}